docs/source/developers/cpp.rst

```diff
@@ -630,6 +630,27 @@ The report is then generated in ``compat_reports/libarrow`` as a HTML.
 
 .. _developers-cpp-windows:
 
+Debugging with Xcode on macOS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Xcode is the IDE provided with macOS and can be use to develop and debug Arrow 
+by generating an Xcode project:
+
+.. code-block:: shell
+
+   cd cpp
+   mkdir xcode-build
+   cd xcode-build
+   cmake .. -G Xcode ^
```

Comment:
We need to use "\\" for shell. "^" is for batch file.

Reply:
Good catch @kou, I'll fix this in a followup PR.